Pull all the tracking event names into one place

zachmargolis opened the pull request:
**Why:**
We have different styles and naming conventions throughout the app,
forcing all these events to be in one places encourages developers
to look and see what we already have before adding something new.

-----

I'll make a second PR to try and make the names consistent. This first PR is just pulling all the events into constant names and putting them in one place (and alphabetizing)

**UPDATE**: I added a second commit to here that makes things consistent. It might be a lot of noise, so I'd recommend looking at the two commits separately to reduce cognitive overhead.

jessieay commented:
I like this idea a lot! Let us know when you want a review (usually ppl on this team do not review WIP PRs)

zachmargolis commented:
@jessieay thanks for clarifying! I removed the "WIP" and will make a second PR with a proposed set of names to make things consistent. Having this at least means it won't get any more merge conflicts, and is straightforward to follow

jessieay commented on app/services/analytics.rb:
```diff
@@ -30,4 +30,45 @@ def ahoy
+  EMAIL_CONFIRMATION_TOKEN_EXPIRED = 'Email Confirmation: token expired'.freeze
+  EMAIL_CONFIRMATION_USER_ALREADY_CONFIRMED = 'Email Confirmation: User Already Confirmed'.freeze
+  EMAIL_CONFIRMATION_VALID_TOKEN = 'Email Confirmation: valid token'.freeze
+  GET_REQUEST = 'GET Request'.freeze
```
I think this should have interpolated vars?

zachmargolis replied:
I removed the interpolation on purpose and moved the values to the options hash.

The tracking code takes the extra arguments in the hash and just records them, I checked with @monfresh and it seems like we'll be able to query our analytics by the values in the hash so string interpolation seemed like it would make things harder to query for (and create more event types overall)

zachmargolis commented:
**UPDATE**: I added a second commit to here that renames a bunch of these events. It might be a lot of noise, so I'd recommend looking at the two commits separately to reduce cognitive overhead (can also split into two PRs if folks desire, but my thinking is that one PR is less likely to get stale and need rebasing)

zachmargolis commented on app/services/analytics.rb:
```diff
@@ -30,4 +30,50 @@ def ahoy
     user.uuid
   end
+
+  # rubocop:disable Metrics/LineLength
```
I disabled line length here because leaving everything as one-liners makes it easy to use my editor to sort these constants alphabetically

zachmargolis commented:
@jessieay re: [your comment about interpolation](https://github.com/18F/identity-idp/pull/628#pullrequestreview-6158447) -- I removed the interpolation intentionally. The tracking library records the hash of options, and many analytics providers should let us query and aggregate based on those.
